libsel4vmmplatsupport/src/ringbuf.c

```diff
@@ -0,0 +1,57 @@
+/*
```

Comment:
I wonder if `libsel4vmmplatsupport` is the best place for such a lib. We already have https://github.com/seL4/projects_libs/tree/master/libringbuffer, seems the new one is a bit more advanced. Would it make sense to add it there?

Reply:
Or alternatively, keep this implementation private to the zynqmp/devices/vuart.c for now